plugins/destination/bigquery/client/read.go

```diff
@@ -0,0 +1,107 @@
+package client
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"google.golang.org/api/iterator"
+)
+
+const (
+	readSQL = "SELECT * FROM `%s.%s.%s` WHERE `_cq_source_name` = @cq_source_name"
+)
+
+func (*Client) createResultsArray(table *schema.Table) []bigquery.Value {
+	results := make([]bigquery.Value, 0, len(table.Columns))
+	for _, col := range table.Columns {
+		switch col.Type {
```

Comment:
Maybe add a `default` case with a safety `panic()`, to blow up on unhandled/new types? Just like the one in `SchemaTypeToBigQuery`